src/shogun/machine/LinearMulticlassMachine.h

```diff
@@ -82,6 +77,42 @@ class LinearMulticlassMachine : public MulticlassMachine
 
 	protected:
 
+		bool train_machine(std::shared_ptr<Features> data) override
+		{
+			m_features = data->as<DotFeatures>();
+			require(m_multiclass_strategy, "Multiclass strategy not set");
+			int32_t num_classes = m_labels->as<MulticlassLabels>()->get_num_classes();
+   			m_multiclass_strategy->set_num_classes(num_classes);
+
+			m_machines.clear();
+			auto train_labels = std::make_shared<BinaryLabels>(get_num_rhs_vectors());
+
+			m_multiclass_strategy->train_start(
+				multiclass_labels(m_labels), train_labels);
+			while (m_multiclass_strategy->train_has_more())
+			{
+				SGVector<index_t> subset=m_multiclass_strategy->train_prepare_next();
+				if (subset.vlen)
+				{
+					train_labels->add_subset(subset);
+					add_machine_subset(subset);
+				}
+
+				m_machine->train(data, train_labels);
+				m_machines.push_back(get_machine_from_trained(m_machine));
+
+				if (subset.vlen)
+				{
+					train_labels->remove_subset();
+					remove_machine_subset();
+				}
+			}
+
+			m_multiclass_strategy->train_stop();
+
+
+			return true;
+		}
```

Comment:
where did this come from?

Reply:
this part from `MulticlassMachine`, as now `LinearMachine` use `train(shared_ptr<Features>, shared_ptr<Labels>)`, i add a override version for `LinearMulticlassMachine`.
https://github.com/shogun-toolbox/shogun/blob/f01b18655994c7a4630acc4b67b60912a957217e/src/shogun/machine/MulticlassMachine.cpp#L217-L257

Reply:
oh i see thanks!